### PR TITLE
Make set and map types aliases flat and steady

### PR DESCRIPTION
**Pull request by woelke**
In this PR we replace the type of the  aliases `set` and `map` by a flat steady_set and steady_map because (i) data is immutable, (ii) it is a more cache-friendly data structure, and (iii) set cardinalities are typically small where sequential brute-force access/find is fast enough.

The first commit replaces the type of `set` and the second commit replaces the type of `map`.

### CODE REVIEW

File: libvast/vast/aliases.hpp [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include <set>
```
Comment (mavam) [inline libvast/vast/aliases.hpp]: This include can be removed now.

File: libvast/vast/detail/vector_set.hpp [hunk 2]
```diff
@@ -196,9 +194,27 @@ class vector_set : totally_ordered<vector_set<T, Allocator, Policy>> {
...
+template <class T, class Allocator, class Policy>
+struct hash<vast::detail::vector_set<T, Allocator, Policy>> {
+  using vector_type = vast::detail::vector_set<T, Allocator, Policy>;
+  size_t operator()(const vector_type& x) const {
+    return vast::uhash<vast::xxhash>{}(
+      static_cast<const typename vector_type::vector_type&>(x));
+  }
+};
+
+} // namespace std
```
Comment (mavam) [inline libvast/vast/detail/vector_set.hpp]: I don't think we don't need a specialization of `std::hash`.

File: libvast/vast/detail/vector_set.hpp [hunk 3]
```diff
@@ -196,9 +194,27 @@ class vector_set : totally_ordered<vector_set<T, Allocator, Policy>> {
...
+
```
Comment (mavam) [inline libvast/vast/detail/vector_set.hpp]: No trailing newlines.

File: libvast/vast/detail/vector_set.hpp [hunk 4]
```diff
@@ -196,9 +194,27 @@ class vector_set : totally_ordered<vector_set<T, Allocator, Policy>> {
     return xs_;
   }
 
+  template <class Inspector>
+  friend auto inspect(Inspector&f, vector_set<T, Allocator, Policy>& x) {
```
Comment (mavam) [inline libvast/vast/detail/vector_set.hpp]: This can be shortened to

```cpp
friend auto inspect(Inspector&f, vector_set& x) {
```

Comment (mavam) [reply libvast/vast/detail/vector_set.hpp]: Also, a set is a container. We usually use `xs` and not `x` for that.

File: libvast/vast/detail/vector_set.hpp [hunk 5]
```diff
@@ -16,6 +16,9 @@
 #include <algorithm>
 #include <vector>
 
+#include "vast/concept/hashable/uhash.hpp"
+#include "vast/concept/hashable/xxhash.hpp"
+
```
Comment (mavam) [inline libvast/vast/detail/vector_set.hpp]: Why do we need these?

File: libvast/vast/aliases.hpp [hunk 6]
```diff
@@ -15,12 +15,11 @@
 
 #include <cstdint>
 #include <limits>
-#include <map>
 #include <memory>
-#include <set>
```
Comment (mavam) [inline libvast/vast/aliases.hpp]: This commit seems to do more than what it describes. In the future, please keep the commits more-or-less contained or point out non-related changes.

File: libvast/vast/detail/vector_set.hpp (hunk 5 above)
Comment (woelke) [reply libvast/vast/detail/vector_set.hpp]: Oh yes, these includes are outdated.

**Pull request merged**